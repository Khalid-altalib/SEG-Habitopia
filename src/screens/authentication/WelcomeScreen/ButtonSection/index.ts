import ButtonSection from "./ButtonSection";
export default ButtonSection;
