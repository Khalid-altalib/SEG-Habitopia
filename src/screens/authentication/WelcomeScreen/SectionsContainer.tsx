// React
import { ReactNode } from "react";

// React Native
import { StyleProp, View, ViewStyle } from "react-native";

/**
 * A container for separating sections. This is ideal for use with the title and button containers.
 *
 * @param props The properties passed to the container.
 * @returns The component representing the container.
 */
const SectionsContainer = ({ children }: Props): JSX.Element => (
  <View style={containerStyle}>{children}</View>
);

/**
 * The properties passed to the container.
 */
type Props = {
  /** The sections to show within the container. */
  children: ReactNode[];
};

/**
 * A style which makes the container take up all space available to it, add spacing only between
 * sections, and add vertical padding.
 */
const containerStyle: StyleProp<ViewStyle> = {
  /** Take up all space available */
  width: "100%",
  height: "100%",
  /** Add spacing between sections */
  justifyContent: "space-between",
  /** Add vertical padding */
  paddingTop: "30%",
  paddingBottom: "20%",
};

export default SectionsContainer;
