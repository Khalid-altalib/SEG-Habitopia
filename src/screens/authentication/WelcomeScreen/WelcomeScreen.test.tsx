// React
import React from "react";

// React Native Testing Library
import { render, screen } from "@testing-library/react-native";

// Redux
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

// Habitopia
import WelcomeScreen from "@screens/authentication/WelcomeScreen";
import TestingWrapper from "@app/testingWrapper";

it("Renders correctly.", () => {
  const mockStore = configureStore([thunk])({});

  render(
    <TestingWrapper store={mockStore}>
      <WelcomeScreen />
    </TestingWrapper>
  );

  expect(screen.toJSON()).toMatchSnapshot();
});
