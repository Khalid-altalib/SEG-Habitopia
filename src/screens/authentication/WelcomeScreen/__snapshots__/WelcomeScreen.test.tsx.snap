// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly. 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <ViewManagerAdapter_ExpoLinearGradient
    proxiedProperties={
      {
        "colors": [
          4293190887,
          4294243573,
        ],
        "endPoint": undefined,
        "locations": undefined,
        "startPoint": undefined,
      }
    }
    style={
      {
        "height": "100%",
      }
    }
  >
    <View
      style={
        {
          "height": "100%",
          "padding": 25,
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "height": "100%",
            "justifyContent": "space-between",
            "paddingBottom": "20%",
            "paddingTop": "30%",
            "width": "100%",
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
            }
          }
        >
          <Text
            style={
              [
                {
                  "color": "#18181b",
                },
                {
                  "fontFamily": undefined,
                  "fontSize": 40,
                },
                {
                  "marginBottom": "3%",
                },
              ]
            }
          >
            Habitopia.
          </Text>
          <Text
            style={
              [
                {
                  "color": "#18181b",
                },
                {
                  "color": "#71717a",
                  "fontFamily": undefined,
                  "fontSize": 15,
                },
                {
                  "marginBottom": "3%",
                },
              ]
            }
          >
            Unlimited accountability to beat procrastination.
          </Text>
        </View>
        <View>
          <View
            isFullWidth={true}
            onPress={[Function]}
            style={
              [
                {
                  "borderRadius": 10,
                  "shadowColor": "black",
                  "shadowOffset": {
                    "height": 5,
                    "width": 5,
                  },
                  "shadowOpacity": 0.15,
                },
              ]
            }
            type={0}
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <ViewManagerAdapter_ExpoLinearGradient
                proxiedProperties={
                  {
                    "colors": [
                      4278220774,
                      4286456526,
                    ],
                    "endPoint": [
                      1,
                      1,
                    ],
                    "locations": undefined,
                    "startPoint": [
                      0,
                      0,
                    ],
                  }
                }
                style={
                  [
                    {
                      "alignItems": "center",
                      "borderRadius": 10,
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "paddingHorizontal": 40,
                      "paddingVertical": 15,
                    },
                    {
                      "width": "100%",
                    },
                  ]
                }
              >
                <Text
                  style={
                    [
                      {
                        "color": "#18181b",
                      },
                      {
                        "color": "#e4e4e7",
                        "fontFamily": undefined,
                        "fontSize": 19,
                      },
                      [
                        null,
                        {
                          "alignSelf": "center",
                        },
                      ],
                    ]
                  }
                >
                  Create account
                </Text>
                <View
                  style={
                    {
                      "marginLeft": 5,
                    }
                  }
                />
              </ViewManagerAdapter_ExpoLinearGradient>
            </View>
          </View>
          <View
            isFullWidth={true}
            onPress={[Function]}
            style={
              {
                "marginTop": 20,
              }
            }
            type={1}
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "marginTop": 20,
                  "opacity": 1,
                }
              }
            >
              <ViewManagerAdapter_ExpoLinearGradient
                proxiedProperties={
                  {
                    "colors": [
                      4288782762,
                      4285624698,
                    ],
                    "endPoint": [
                      1,
                      1,
                    ],
                    "locations": undefined,
                    "startPoint": [
                      0,
                      0,
                    ],
                  }
                }
                style={
                  [
                    {
                      "alignItems": "center",
                      "borderRadius": 10,
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "paddingHorizontal": 40,
                      "paddingVertical": 15,
                    },
                    {
                      "width": "100%",
                    },
                  ]
                }
              >
                <Text
                  style={
                    [
                      {
                        "color": "#18181b",
                      },
                      {
                        "color": "#e4e4e7",
                        "fontFamily": undefined,
                        "fontSize": 19,
                      },
                      [
                        null,
                        {
                          "alignSelf": "center",
                        },
                      ],
                    ]
                  }
                >
                  Sign in
                </Text>
                <View
                  style={
                    {
                      "marginLeft": 5,
                    }
                  }
                />
              </ViewManagerAdapter_ExpoLinearGradient>
            </View>
          </View>
        </View>
      </View>
    </View>
  </ViewManagerAdapter_ExpoLinearGradient>
</RNCSafeAreaProvider>
`;
